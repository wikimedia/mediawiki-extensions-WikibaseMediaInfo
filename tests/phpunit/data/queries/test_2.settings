{
	"title": "Entities found, with lang fallbacks, no stemming",
	"term": "another search term",
	"entityIdsForTerm": {
		"another search term": [
			"Q9",
			"Q2"
		]
	},
	"defaultProperties": [
		"P999"
	],
	"userLanguage": "ga",
	"fallbackLangs": [ "fr", "es" ],
	"settings": {
		"statement": 50.0,
		"descriptions": 100.0,
		"all": 0.09,
		"category": 0.07,
		"title": 0.06,
		"redirect.title": 0.05,
		"text": 0.04,
		"auxiliary_text": 0.03,
		"file_text": 0.02
	}
}
