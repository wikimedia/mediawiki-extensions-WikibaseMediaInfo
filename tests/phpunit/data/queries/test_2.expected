{
    "bool": {
        "must": [
            {
                "function_score": {
                    "query": {
                        "bool": {
                            "minimum_should_match": 1,
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 1,
                                        "queries": [
                                            {
                                                "bool": {
                                                    "minimum_should_match": 0,
                                                    "filter": [
                                                        {
                                                            "multi_match": {
                                                                "query": "another search term",
                                                                "fields": [
                                                                    "all",
                                                                    "all.plain"
                                                                ],
                                                                "operator": "and"
                                                            }
                                                        }
                                                    ],
                                                    "should": [
                                                        {
                                                            "match": {
                                                                "title": {
                                                                    "query": "another search term",
                                                                    "boost": 0.45
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "title.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.15
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "category": {
                                                                    "query": "another search term",
                                                                    "boost": 0.525
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "category.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.175
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "redirect.title": {
                                                                    "query": "another search term",
                                                                    "boost": 0.375
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "redirect.title.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.125
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "heading": {
                                                                    "query": "another search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "heading.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "auxiliary_text": {
                                                                    "query": "another search term",
                                                                    "boost": 0.225
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "auxiliary_text.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.075
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "text": {
                                                                    "query": "another search term",
                                                                    "boost": 0.3
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "text.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "file_text": {
                                                                    "query": "another search term",
                                                                    "boost": 0.15
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "file_text.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 0.05
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "suggest": {
                                                                    "query": "another search term",
                                                                    "boost": 1
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "descriptions.ga.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 2.5
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "descriptions.fr.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 2.5
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "descriptions.es.plain": {
                                                                    "query": "another search term",
                                                                    "boost": 2.5
                                                                }
                                                            }
                                                        }
                                                    ]
                                                }
                                            }
                                        ]
                                    }
                                },
                                {
                                    "dis_max": {
                                        "queries": [
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P999=Q9",
                                                        "boost": 20
                                                    }
                                                }
                                            },
                                            {
                                                "match": {
                                                    "statement_keywords": {
                                                        "query": "P999=Q2",
                                                        "boost": 10
                                                    }
                                                }
                                            }
                                        ]
                                    }
                                }
                            ]
                        }
                    },
                    "functions": [
                        {
                            "script_score": {
                                "script": {
                                    "source": "100 \/ ( 1 + exp( -1 * ( _score + intercept ) ) )",
                                    "params": {
                                        "intercept": 1
                                    },
                                    "lang": "expression"
                                }
                            }
                        }
                    ]
                }
            }
        ],
        "filter": [
            {
                "terms": {
                    "namespace": [
                        0
                    ]
                }
            }
        ]
    }
}